stdlib/public/core/StringBridge.swift

```diff
@@ -141,43 +51,41 @@ extension String {
   @inline(never) @_semantics("stdlib_binary_only") // Hide the CF dependency
   public // SPI(Foundation)
   init(_cocoaString: AnyObject) {
+    // If the NSString is actually a Swift String in disguise, 
+    // we can just copy out the internal repr.
     if let wrapped = _cocoaString as? _NSContiguousString {
       self._core = wrapped._core
       return
     }
 
-    // "copy" it into a value to be sure nobody will modify behind
-    // our backs.  In practice, when value is already immutable, this
-    // just does a retain.
-    let cfImmutableValue
-      = _stdlib_binary_CFStringCreateCopy(_cocoaString) as AnyObject
-
-    let length = _swift_stdlib_CFStringGetLength(cfImmutableValue)
+    let length = _swift_stdlib_CFStringGetLength(_cocoaString)
 
     // Look first for null-terminated ASCII
-    // Note: the code in clownfish appears to guarantee
-    // nul-termination, but I'm waiting for an answer from Chris Kane
-    // about whether we can count on it for all time or not.
+    // FIXME: we're assuming "null-terminated ASCII" is a synonym for "ASCII". 
+    // This appears to be true, but we should verify that ObjC actually 
+    // guarantees this. Only a perf concern; soundness doesn't rely on this.
     let nulTerminatedASCII = _swift_stdlib_CFStringGetCStringPtr(
-      cfImmutableValue, kCFStringEncodingASCII)
+      _cocoaString, kCFStringEncodingASCII)
 
-    // start will hold the base pointer of contiguous storage, if it
-    // is found.
-    var start: UnsafeMutableRawPointer?
     let isUTF16 = (nulTerminatedASCII == nil)
+    let buffer = _StringBuffer(capacity: length, 
+                               initialSize: length, 
+                               elementWidth: isUTF16 ? 2 : 1)
+
     if isUTF16 {
-      let utf16Buf = _swift_stdlib_CFStringGetCharactersPtr(cfImmutableValue)
-      start = UnsafeMutableRawPointer(mutating: utf16Buf)
+      // If we aren't ascii, ask the NSString to copy itself into our buffer
+      // FIXME(eager-bridging): is the range variant or the non-range variant
+      // more effecient, assuming we've already computed the length?
+      _swift_stdlib_CFStringGetCharacters(
```

Comment:
Ideally this should be the NSString API (the CF version is slower)
But you could call the fast version before jumping to that:

```
@interface NSString (SPI)
- (nullable const unichar *)_fastCharacterContents;
@end
```

Then after calling that use `-[NSString getCharacters:range:]` since that is faster than the CF call